Angular/src/services/gemini-google-ai/gemini-google-ai.service.ts

```diff
@@ -1,16 +1,29 @@
 import { Injectable } from '@angular/core';
 import { GoogleGenerativeAI } from '@google/generative-ai';
+import { environment } from '../../environments/environment';
+import { asyncScheduler, from, map, Observable, of, scheduled } from 'rxjs';
 
 @Injectable({
   providedIn: 'root'
 })
 export class GeminiGoogleAiService {
 
-  constructor(
-    private genAi: GoogleGenerativeAI
-  ) { }
+  // instance to initiate Gemini - Google Ai with API_KEY
+  private genAI: GoogleGenerativeAI;
 
-  geminiText(){
-    // get data from gemini using text prompt
+  constructor() {
+    this.genAI = new GoogleGenerativeAI(environment.googleAiKey);
+  }
+
+  /**
+   * Communicate with Gemini - Google Ai using text prompt
+   */
+  geminiText(prompt: string): Observable<string> {
+    const model: any = this.genAI.getGenerativeModel({ model: "gemini-1.5-flash" });
+
+    // converting Promise to Observable - latest way "After Deprecation"
+    return scheduled(from(model.generateContent(prompt)), asyncScheduler).pipe(
```

Comment:
earlier it was only using `from `but when i hover it says it will be not working with next v8. if you want i am able to do with older version, or we can remain it's simple usage using promise rather than observable. is it ok?

Reply:
Let's leave it for now and revisit it later, maybe.

Reply:
this is the simple way of doing it.

`return from(model.generateContent(prompt)).pipe(
      map(result => result.response.text())
    );`

but it says, 

@deprecated — The scheduler parameter will be removed in v8. Use scheduled. 
Details: https://rxjs.dev/deprecations/scheduler-argument

Reply:
Keep what you have for now.

Reply:
thanks.